### PR TITLE
flight projection (no history)

### PR DESCRIPTION
**Pull request by massimosiani**

### CODE REVIEW

File: src/main/scala/it/bitrock/dvs/streams/topologies/FlightInterpolatedListStream.scala [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+  private val currentSnapshot = "currentSnapshot"
+
+  def buildTopology(config: AppConfig, kafkaStreamsOptions: KafkaStreamsOptions): List[(Topology, Properties)] = {
+    implicit val stringKeySerde: Serde[String]                           = kafkaStreamsOptions.stringKeySerde
+    implicit val flightReceivedListEventSerde: Serde[FlightReceivedList] = kafkaStreamsOptions.flightReceivedListEventSerde
+
+    val streamsBuilder = new StreamsBuilder
+
+    val stateStore = Stores.keyValueStoreBuilder(
+      Stores.persistentKeyValueStore("FlightListInterpolationStateStore"),
```
Comment (simoexpo) [inline src/main/scala/it/bitrock/dvs/streams/topologies/FlightInterpolatedListStream.scala]: I'd extract the name in a constant

Comment (lodamar) [reply src/main/scala/it/bitrock/dvs/streams/topologies/FlightInterpolatedListStream.scala]: Yep, I was thinking about the same thing even if it's used only one time

File: src/main/scala/it/bitrock/dvs/streams/topologies/FlightInterpolatedListStream.scala [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    List((streamsBuilder.build(props), props))
+  }
+
+  @SuppressWarnings(Array("DisableSyntax.var"))
+  private def interpolationTransformer(
+      stateStoreName: String,
+      interpolationInterval: FiniteDuration
+  ): Transformer[String, FlightReceivedList, KeyValue[String, FlightReceivedList]] =
+    new Transformer[String, FlightReceivedList, KeyValue[String, FlightReceivedList]] {
+      private var processorContext: ProcessorContext                       = _
```
Comment (massimosiani) [inline src/main/scala/it/bitrock/dvs/streams/topologies/FlightInterpolatedListStream.scala]: The field isn't used. Can we throw it away or it enforces side effects?

File: src/test/scala/it/bitrock/dvs/streams/topologies/FlightInterpolatedListStreamSpec.scala [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    with AnyWordSpecLike
+    with EmbeddedKafkaStreams
+    with OptionValues
+    with TestValues {
+  "FlightInterpolatedListStream" should {
+    "produce interpolated record of FlightReceivedList in interpolated topic" in ResourceLoaner.withFixture {
+      case Resource(embeddedKafkaConfig, appConfig, kafkaStreamsOptions, _) =>
+        implicit val embKafkaConfig: EmbeddedKafkaConfig                     = embeddedKafkaConfig
+        implicit val keySerde: Serde[String]                                 = kafkaStreamsOptions.stringKeySerde
+        implicit val flightReceivedListEventSerde: Serde[FlightReceivedList] = kafkaStreamsOptions.flightReceivedListEventSerde
```
Comment (massimosiani) [inline src/test/scala/it/bitrock/dvs/streams/topologies/FlightInterpolatedListStreamSpec.scala]: unused

File: src/test/scala/it/bitrock/dvs/streams/topologies/FlightInterpolatedListStreamSpec.scala [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+          )
+
+          messages(appConfig.kafka.topology.flightInterpolatedListTopic.name)
+
+        }
+
+        val endTestTime = Instant.now.toEpochMilli
+
+        receivedRecords should have size expectedMessages
+        receivedRecords.tail.foreach {
```
Comment (massimosiani) [inline src/test/scala/it/bitrock/dvs/streams/topologies/FlightInterpolatedListStreamSpec.scala]: the head here should be the original set of messages I guess

File: src/test/scala/it/bitrock/dvs/streams/topologies/FlightInterpolatedListStreamSpec.scala [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+          case (k, v) =>
+            k.toLong shouldBe <(endTestTime)
+            k.toLong shouldBe >(now.minus(1, ChronoUnit.SECONDS).toEpochMilli)
+
+            val confrontedFlights = v.elements.sortBy(_.icaoNumber) zip receivedList.elements.sortBy(_.icaoNumber)
+
+            confrontedFlights.foreach { case (interpolated, original) => interpolated.geography should not be original.geography }
```
Comment (massimosiani) [inline src/test/scala/it/bitrock/dvs/streams/topologies/FlightInterpolatedListStreamSpec.scala]: let's check a bit more properties

File: src/main/scala/it/bitrock/dvs/streams/topologies/FlightInterpolatedListStream.scala (hunk 2 above)
Comment (lodamar) [reply src/main/scala/it/bitrock/dvs/streams/topologies/FlightInterpolatedListStream.scala]: Good catch, an oversight of a previous implementation

File: src/test/scala/it/bitrock/dvs/streams/topologies/FlightInterpolatedListStreamSpec.scala (hunk 4 above)
Comment (lodamar) [reply src/test/scala/it/bitrock/dvs/streams/topologies/FlightInterpolatedListStreamSpec.scala]: yes, we don't need to check first message

File: src/test/scala/it/bitrock/dvs/streams/topologies/FlightInterpolatedListStreamSpec.scala (hunk 5 above)
Comment (lodamar) [reply src/test/scala/it/bitrock/dvs/streams/topologies/FlightInterpolatedListStreamSpec.scala]: good work 👍

**Pull request merged**